DNN Platform/DotNetNuke.Web/Startup.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DotNetNuke.DependencyInjection;
+using DotNetNuke.DependencyInjection.Extensions;
+using DotNetNuke.Instrumentation;
+using DotNetNuke.Web.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace DotNetNuke.Web
+{
+    public class Startup : IDnnStartup
+    {
+        private static readonly ILog _logger = LoggerSource.Instance.GetLogger(typeof(Startup));
+        public Startup()
+        {
+            Configure();
+        }
+
+        private void Configure()
+        {
+            var services = new ServiceCollection();
+            ConfigureServices(services);
+            DependencyProvider = services.BuildServiceProvider();
+        }
+
+        public IServiceProvider DependencyProvider { get; private set; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var startupTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .Where(x => x != Assembly.GetAssembly(typeof(Startup)))
+                .SelectMany(x => x.SafeGetTypes())
+                .Where(x => typeof(IDnnStartup).IsAssignableFrom(x) &&
+                            x.IsClass &&
+                            !x.IsAbstract);
+
+            var startupInstances = startupTypes
+                .Select(x => (IDnnStartup)Activator.CreateInstance(x));
```

Comment:
We probably need to wrap this in a `try`/`catch`, in case there's a bug in someone's constructor code, we don't want to take the website down.

Reply:
@ahoefling this is my only remaining issue/question before I'm ready to approve (just wanted to make sure you didn't miss it since I added while you were reviewing previous comments)

Reply:
In this scenario let's say someone creates a `Startup` file that implements `IDnnStartup` in their 3rd-party library. They decide to go against the best practices (which happens) and have some code in the constructor that causes an exception. Placing a try-catch around the activator code will prevent the website from crashing.

Yes this makes sense to me, I'll add it in. Great feedback!

Reply:
I added this comment on the `Select` call, but the `try` would need to be around the `foreach`, FYI.  Wanted to make sure I didn't throw you off track.

Reply:
I updated the code and added a try-catch for both the foreach and the select